### ISSUE TITLE
Memory error. _targets/meta/meta grows too big

### ISSUE THREAD

**Issue by gorkang** [wlandau/targets#186]
## Prework

* [x] Read and agree to the [code of conduct](https://github.com/wlandau/targets/blob/master/CODE_OF_CONDUCT.md) and [contributing guidelines](https://github.com/wlandau/targets/blob/master/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/wlandau/targets/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [x] Post a [minimal reproducible example](https://www.tidyverse.org/help/) so the maintainer can troubleshoot the problems you identify. A reproducible example is:
    * [x] **Runnable**: post enough R code and data so any onlooker can create the error on their own computer.
    * [ ] **Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.
    * [ ] **Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Description

Sometimes, when working in a project. Suddenly I start receiving the error you can see below. Poking around, I can see _targets/meta/meta has grown to > 1GB.

Specifically, the `error` column of the meta DF seems to contain the problematic content.

## Reproducible example

I am attaching here a reduced version of the project where it happens: [memory_error.zip](https://github.com/wlandau/targets/files/5339619/memory_error.zip)

If you delete the _targets folder (or tar_destroy() it), the project runs again.

* [X] Post a [minimal reproducible example](https://www.tidyverse.org/help/) so the maintainer can troubleshoot the problems you identify. A reproducible example is:
    * [x] **Runnable**: post enough R code and data so any onlooker can create the error on their own computer.
    * [ ] **Minimal**: reduce runtime wherever possible and remove complicated details that are irrelevant to the issue at hand.
    * [ ] **Readable**: format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Expected result

meta should not grow so much as to cause an error. Alternatively, if meta grows that much, a specific warning about it would be nice.

## Diagnostic information

```
targets::tar_make()
Error in data.table::fwrite(x = data, file = tmp, sep = database_sep_outer,  : 
  Unable to allocate -138142212 MB * 8 thread buffers; '12: Cannot allocate memory'. Please read ?fwrite for nThread, buffMB and verbose options.
Error: callr subprocess failed: Unable to allocate -138142212 MB * 8 thread buffers; '12: Cannot allocate memory'. Please read ?fwrite for nThread, buffMB and verbose options.
Type .Last.error.trace to see where the error occured
> .Last.error.trace

 Stack trace:

 Process 25034:
 1. targets::tar_make()
 2. targets:::callr_outer(targets_function = tar_make_inner, targets_arguments = targets_arguments,  ...
 3. targets:::trn(is.null(callr_function), callr_inner(target_script_path(),  ...
 4. base:::do.call(callr_function, prepare_callr_arguments(callr_function,  ...
 5. (function (func, args = list(), libpath = .libPaths(), repos = default_repos(),  ...
 6. callr:::get_result(output = out, options)
 7. throw(newerr, parent = remerr[[2]])

 x callr subprocess failed: Unable to allocate -138142212 MB * 8 thread buffers; '12: Cannot allocate memory'. Please read ?fwrite for nThread, buffMB and verbose options. 

 Process 25091:
 19. (function (targets_script, targets_function, targets_arguments)  ...
 20. base:::source(targets_script)
 21. base:::withVisible(eval(ei, envir))
 22. base:::eval(ei, envir)
 23. base:::eval(ei, envir)
 24. targets::tar_invalidate(matches("TESTS"))
 25. meta$database$overwrite_storage(data)
 26. data.table::fwrite(x = data, file = tmp, sep = database_sep_outer,  ...
 27. base:::.handleSimpleError(function (e)  ...
 28. h(simpleError(msg, call))

 x Unable to allocate -138142212 MB * 8 thread buffers; '12: Cannot allocate memory'. Please read ?fwrite for nThread, buffMB and verbose options. 
```

**Comment by gorkang** [comment]
Specifically, the 8th row of the error column in the meta file has the 1.1GB content. 

I saved that content into a plain text file. Doing `cat bad_file` seems to give me 1.1GB of `"`.

**Comment by gorkang** [comment]
That 8th line refers to one of the functions I commented out and deleted from _targets.R when trying to reduce the complexity of the project (check_trialid_question.R). Here the function attached: [check_trialid_question.zip](https://github.com/wlandau/targets/files/5339899/check_trialid_question.zip)

**Comment by wlandau** [comment]
Thanks for sharing so much detail. Looks like the error of `check_trial_question_DF` has a 1073741872-character error message. We should definitely shorten errors and warnings. Should be an easy fix.

**Comment by baslat** [comment]
@wlandau I've noticed this same issue this week. Using targets >= 1.8.0 (I recently updated to 1.11.3 as part of debugging, so I have used various versions), my meta file grew to about 40GB. It was indeed the error column, with about many rows each with 2 million character error messages, which were mostly `"`. I have `tar_option_set(error = null)`. Once I get my pipeline working again I will try to make a reprex if that would be helpful.
